### PR TITLE
Cleanup from transaction migration

### PR DESCRIPTION
**Pull request by zachmu**
First batch of cleanup, getting rid of uses of RepoStateWriter as much as possible.

To push this farther requires new abstractions that can handle updating the HEAD, head commit, and working roots at the same time, which is the next step.

### CODE REVIEW

File: go/libraries/doltcore/dtestutils/testcommands/command.go [hunk 1]
```diff
@@ -48,7 +48,10 @@ func (a StageAll) Exec(t *testing.T, dEnv *env.DoltEnv) error {
 	roots, err := dEnv.Roots(context.Background())
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/dtestutils/testcommands/command.go:48]: we need to kill off this package, I added `testcommands` before realizing we could just use the actual commands in tests

File: go/libraries/doltcore/env/actions/staged.go [hunk 2]
```diff
@@ -16,69 +16,44 @@ package actions
...
-		return err
-	}
-
+func StageTables(ctx context.Context, roots doltdb.Roots, docs doltdocs.Docs, tbls []string) (doltdb.Roots, error) {
 	if len(docs) > 0 {
+		var err error
 		roots.Working, err = doltdocs.UpdateRootWithDocs(ctx, roots.Working, docs)
 		if err != nil {
-			return err
+			return doltdb.Roots{}, nil
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/env/actions/staged.go:29]:
```suggestion
			return doltdb.Roots{}, err
```

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/env/actions/staged.go:29]: we should added Lars' linter

File: go/libraries/doltcore/env/environment.go [hunk 3]
```diff
@@ -1016,3 +944,59 @@ func (dEnv *DoltEnv) GetUserHomeDir() (string, error) {
 func (dEnv *DoltEnv) TempTableFilesDir() string {
 	return mustAbs(dEnv, dEnv.GetDoltDir(), tempTablesDir)
 }
+
+// GetGCKeepers returns the hashes of all the objects in the environment provided that should be perserved during GC.
+func GetGCKeepers(ctx context.Context, env *DoltEnv) ([]hash.Hash, error) {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/env/environment.go:949]: I think that we don't need this anymore. Working and stage are stored in a `dataset` now, which should protect them from GC. That said, I think we should make that a separate change

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM

File: go/libraries/doltcore/dtestutils/testcommands/command.go (hunk 1 above)
Comment (zachmu) [reply go/libraries/doltcore/dtestutils/testcommands/command.go:48]: added TODO

File: go/libraries/doltcore/env/actions/staged.go (hunk 2 above)
Comment (zachmu) [reply go/libraries/doltcore/env/actions/staged.go:29]: fixed

File: go/libraries/doltcore/env/environment.go (hunk 3 above)
Comment (zachmu) [reply go/libraries/doltcore/env/environment.go:949]: Added a todo

**Pull request merged**